### PR TITLE
Initial CNC integration

### PR DESCRIPTION
**Pull request by pentacular**

### CODE REVIEW

File: ui/app/App.js [hunk 1]
```diff
@@ -1298,6 +1334,29 @@ class App extends React.Component {
...
+      const tabNode = tabsetNode.getChildren()[selected];
+      if (tabNode.id === 'Make/Tabset/Tab') {
+        return;
+      }
+      const { ip, type } = tabNode.getConfig();
+      const cnc = getCnc({ ip, type });
+      if (path && !data) {
+        data = await readOrWatch(path, { workspace });
+      }
+      await cnc.uploadAndRun(data);
```
Comment (BarbourSmith) [inline ui/app/App.js:1357]: 🥇

**Review by BarbourSmith — APPROVED** [review]
This looks awesome!

Are you at all worried about turning off CORS from a security perspective?

File: ui/app/App.js (hunk 1 above)
Comment (pentacular) [reply ui/app/App.js:1357]: I don't think there are any security implications.

The CNC side hasn't changed, and the browser is sending blind, and listening for feedback from the websocket.

It just makes the conversations a bit more awkward.

Comment (BarbourSmith) [reply ui/app/App.js:1357]: Couldn't some random website take control of your CNC and turn it on?

**Pull request merged**